app/views/boards/show.html.erb

```diff
@@ -6,29 +6,56 @@
   Board: <%= @board.title %>
 </div>
 
-<% @lists.each do |list| %>
-
-  <div>
-    <%= list.title %>
-    <%= list.id %>
-    <%= link_to 'Edytuj', edit_board_list_path(list.board, list) %>
-    <%= link_to 'Usuń', board_list_path(list.board, list), method: :delete, data: {confirm: 'Na pewno chcesz usunąć listę?'} %>
-    <%= link_to 'Dodaj ToDoPoint', new_list_todopoint_path(list) %>
-  </div>
+<%= link_to 'TOGGLE_EDIT_MODE', board_path(@board, edit_mode: !@editmode), class: 'a-toggle' %>
 
-  <h4>Todopoints:</h4>
-  <% list.todopoints.where(todopoints: { done: @done_filter}).each do |todopoint| %>
-  <div>
-    <%= @done_filter ? 'Wykonane' : 'Niewykonane' %>
-  </div>
-  <div>
-    <%= link_to "#{todopoint.body}", list_todopoint_path(list, todopoint) %>
-    <%= link_to 'Edytuj', edit_list_todopoint_path(list, todopoint) %>
-    <%= render partial: "shared/todopoint_done_form_#{!@done_filter}", locals: {list: list, todopoint: todopoint} %>
-    <%= link_to 'Usuń', list_todopoint_path(list, todopoint), method: :delete, data: {confirm: 'Na pewno chcesz usunąć ToDiPointa?'} %>
-  </div>
+<% @lists.each_with_index do |list, index| %>
+  <div style="border: 1px solid black; margin-top: 5px" >
+
+    <div>
+      <%= list.title %>
+      <%= link_to 'Edytuj', edit_board_list_path(list.board, list) %>
+      <%= link_to 'Usuń', board_list_path(list.board, list), method: :delete, data: {confirm: 'Na pewno chcesz usunąć listę?'} %>
+      <%= link_to 'Dodaj ToDoPoint', new_list_todopoint_path(list) %>
+    </div>
+
+    <h4>Todopoints:</h4>
+
+    <% list.todopoints.where(todopoints: { done: @done_filter}).order(position: :asc).each do |todopoint| %>
```

Comment:
avoid database connections from view. setup the instance variable @todopoints in controller